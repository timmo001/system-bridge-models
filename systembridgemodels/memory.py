# generated by datamodel-codegen:
#   filename:  memory.json

from __future__ import annotations

from pydantic import BaseModel, Extra, Field


class LastUpdated(BaseModel):
    """
    Last updated
    """

    swap_total: float | None = None
    swap_used: float | None = None
    swap_free: float | None = None
    swap_percent: float | None = None
    swap_sin: float | None = None
    swap_sout: float | None = None
    virtual_total: float | None = None
    virtual_available: float | None = None
    virtual_percent: float | None = None
    virtual_used: float | None = None
    virtual_free: float | None = None


class Memory(BaseModel):
    """
    Memory
    """

    class Config:
        extra = Extra.allow

    id: str | None = Field(None, description="Event ID")
    swap_total: int | None = None
    swap_used: int | None = None
    swap_free: float | None = None
    swap_percent: float | None = None
    swap_sin: int | None = None
    swap_sout: int | None = None
    virtual_total: int | None = None
    virtual_available: int | None = None
    virtual_percent: float | None = None
    virtual_used: int | None = None
    virtual_free: int | None = None
    last_updated: LastUpdated | None = Field(None, description="Last updated")
