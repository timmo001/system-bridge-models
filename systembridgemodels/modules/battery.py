"""Battery."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Battery:
    """Battery."""

    is_charging: bool | None = None
    percentage: float | None = None
