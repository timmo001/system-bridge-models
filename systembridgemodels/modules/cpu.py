"""CPU."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class CPUFrequency:
    """CPU Frequency."""

    current: float | None = None
    min: float | None = None
    max: float | None = None


@dataclass
class CPUStats:
    """CPU Stats."""

    ctx_switches: int | None = None
    interrupts: int | None = None
    soft_interrupts: int | None = None
    syscalls: int | None = None


@dataclass
class CPUTimes:
    """CPU Times."""

    user: float | None = None
    system: float | None = None
    idle: float | None = None
    interrupt: float | None = None
    dpc: float | None = None


@dataclass
class CPU:
    """CPU."""

    count: int | None = None
    frequency: CPUFrequency | None = None
    frequency_per_cpu: list[CPUFrequency] | None = None
    load_average: float | None = None
    power_package: float | None = None
    power_per_cpu: list[float] | None = None
    stats: CPUStats | None = None
    temperature: float | None = None
    times: CPUTimes | None = None
    times_per_cpu: list[CPUTimes] | None = None
    times_percent: CPUTimes | None = None
    times_percent_per_cpu: list[CPUTimes] | None = None
    usage: float | None = None
    usage_per_cpu: list[float] | None = None
    voltage: float | None = None
