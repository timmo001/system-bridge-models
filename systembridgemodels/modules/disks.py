"""Disks."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class DiskIOCounters:
    """Disk IO Counters."""

    read_count: int
    write_count: int
    read_bytes: int
    write_bytes: int
    read_time: int
    write_time: int


@dataclass
class DiskUsage:
    """Disk Usage."""

    total: int
    used: int
    free: int
    percent: float


@dataclass
class DiskPartition:
    """Disk Partition."""

    device: str
    mount_point: str
    filesystem_type: str
    options: str
    max_file_size: int
    max_path_length: int
    usage: DiskUsage | None = None


@dataclass
class Disk:
    """Disk."""

    name: str
    partitions: list[DiskPartition]
    io_counters: DiskIOCounters | None = None


@dataclass
class Disks:
    """Disks."""

    devices: list[Disk]
    io_counters: DiskIOCounters | None = None
