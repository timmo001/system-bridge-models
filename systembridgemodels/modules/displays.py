"""Displays."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class Display:
    """Display."""

    name: str
    resolution_horizontal: int
    resolution_vertical: int
    x: int
    y: int
    width: int | None = None
    height: int | None = None
    is_primary: bool | None = None
    pixel_clock: float | None = None
    refresh_rate: float | None = None


type Displays = list[Display]
