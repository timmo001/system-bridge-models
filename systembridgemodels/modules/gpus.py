"""GPUs."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class GPU:
    """GPU."""

    name: str
    core_clock: float | None = None
    core_load: float | None = None
    fan_speed: float | None = None
    memory_clock: float | None = None
    memory_load: float | None = None
    memory_free: float | None = None
    memory_used: float | None = None
    memory_total: float | None = None
    power_usage: float | None = None
    temperature: float | None = None


type GPUs = list[GPU]
