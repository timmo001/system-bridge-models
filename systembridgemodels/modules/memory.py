"""Memory."""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class MemorySwap:
    """Memory Swap."""

    total: int | None = None
    used: int | None = None
    free: float | None = None
    percent: float | None = None
    sin: int | None = None
    sout: int | None = None


@dataclass
class MemoryVirtual:
    """Memory Virtual."""

    total: int | None = None
    available: int | None = None
    percent: float | None = None
    used: int | None = None
    free: int | None = None
    active: int | None = None
    inactive: int | None = None
    buffers: int | None = None
    cached: int | None = None
    wired: int | None = None
    shared: int | None = None


@dataclass
class Memory:
    """Memory."""

    swap: MemorySwap
    virtual: MemoryVirtual
