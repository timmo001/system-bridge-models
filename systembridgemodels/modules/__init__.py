"""Modules."""
from __future__ import annotations

from dataclasses import dataclass
from enum import Enum

from .battery import Battery
from .cpu import CPU
from .disks import Disks
from .displays import Displays
from .gpus import GPUs
from .media import Media
from .memory import Memory
from .networks import Networks
from .processes import Processes
from .sensors import Sensors
from .system import System


class DataEnum(Enum):
    """Data Enum."""

    BATTERY = "battery"
    CPU = "cpu"
    DISKS = "disks"
    DISPLAYS = "displays"
    GPUS = "gpus"
    MEDIA = "media"
    MEMORY = "memory"
    NETWORKS = "networks"
    PROCESSES = "processes"
    SENSORS = "sensor"
    SYSTEM = "system"


@dataclass
class GetData:
    """Get Data."""

    modules: list[str]


@dataclass
class RegisterDataListener(GetData):
    """Register Data Listener."""


@dataclass
class ModulesData:
    """Data."""

    battery: Battery | None = None
    cpu: CPU | None = None
    disks: Disks | None = None
    displays: Displays | None = None
    gpus: GPUs | None = None
    media: Media | None = None
    memory: Memory | None = None
    networks: Networks | None = None
    processes: Processes | None = None
    sensors: Sensors | None = None
    system: System | None = None
