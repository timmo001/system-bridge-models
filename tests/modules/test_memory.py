"""Test the memory module model."""

from dataclasses import asdict

from systembridgemodels.fixtures.modules.memory import FIXTURE_MEMORY
from systembridgemodels.modules.memory import Memory, MemorySwap, MemoryVirtual


def test_memory(memory: Memory = FIXTURE_MEMORY):
    """Test the memory model."""
    assert isinstance(memory, Memory)
    assert isinstance(memory.swap, MemorySwap)
    assert memory.swap.total == 1000
    assert memory.swap.used == 500
    assert memory.swap.free == 500
    assert memory.swap.percent == 50.0
    assert memory.swap.sin == 100
    assert memory.swap.sout == 100
    assert isinstance(memory.virtual, MemoryVirtual)
    assert memory.virtual.total == 1000
    assert memory.virtual.available == 500
    assert memory.virtual.percent == 50.0
    assert memory.virtual.used == 500
    assert memory.virtual.free == 500
    assert memory.virtual.active == 500
    assert memory.virtual.inactive == 500
    assert memory.virtual.buffers == 500
    assert memory.virtual.cached == 500
    assert memory.virtual.wired == 500
    assert memory.virtual.shared == 500


def test_memory_dict():
    """Test memory dict."""
    memory_dict = asdict(FIXTURE_MEMORY)
    assert isinstance(memory_dict, dict)
    assert isinstance(memory_dict["swap"], dict)
    assert memory_dict["swap"]["total"] == 1000
    assert memory_dict["swap"]["used"] == 500
    assert memory_dict["swap"]["free"] == 500
    assert memory_dict["swap"]["percent"] == 50.0
    assert memory_dict["swap"]["sin"] == 100
    assert memory_dict["swap"]["sout"] == 100
    assert isinstance(memory_dict["virtual"], dict)
    assert memory_dict["virtual"]["total"] == 1000
    assert memory_dict["virtual"]["available"] == 500
    assert memory_dict["virtual"]["percent"] == 50.0
    assert memory_dict["virtual"]["used"] == 500
    assert memory_dict["virtual"]["free"] == 500
    assert memory_dict["virtual"]["active"] == 500
    assert memory_dict["virtual"]["inactive"] == 500
    assert memory_dict["virtual"]["buffers"] == 500
    assert memory_dict["virtual"]["cached"] == 500
    assert memory_dict["virtual"]["wired"] == 500
    assert memory_dict["virtual"]["shared"] == 500

    memory_converted = Memory(**memory_dict)
    test_memory(memory_converted)
