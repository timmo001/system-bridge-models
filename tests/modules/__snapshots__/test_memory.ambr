# serializer version: 1
# name: test_memory
  Memory(swap=MemorySwap(total=1000, used=500, free=500, percent=50.0, sin=100, sout=100), virtual=MemoryVirtual(total=1000, available=500, percent=50.0, used=500, free=500, active=500, inactive=500, buffers=500, cached=500, wired=500, shared=500))
# ---
# name: test_memory_dict
  dict({
    'swap': dict({
      'free': 500,
      'percent': 50.0,
      'sin': 100,
      'sout': 100,
      'total': 1000,
      'used': 500,
    }),
    'virtual': dict({
      'active': 500,
      'available': 500,
      'buffers': 500,
      'cached': 500,
      'free': 500,
      'inactive': 500,
      'percent': 50.0,
      'shared': 500,
      'total': 1000,
      'used': 500,
      'wired': 500,
    }),
  })
# ---
# name: test_memory_dict[memory-dict-converted]
  Memory(swap=MemorySwap(total=1000, used=500, free=500, percent=50.0, sin=100, sout=100), virtual=MemoryVirtual(total=1000, available=500, percent=50.0, used=500, free=500, active=500, inactive=500, buffers=500, cached=500, wired=500, shared=500))
# ---
